services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/docker_api/_core.py

```diff
@@ -320,8 +320,16 @@ async def is_dynamic_service_running(
                 ]
             }
         )
-
-        return len(dynamic_sidecar_services) == 1
+        if len(sidecar_service_list) != 1:
+            return False
+
+        # check if the any of the tasks for the service is in running state
+        service_id = sidecar_service_list[0]["ID"]
+        service_tasks = await client.tasks.list(filters={"service": f"{service_id}"})
+        for task in service_tasks:
```

Comment:
I guess in this case that should be ok to do it this way since there is only one task at a time. you could also add in the filters: ```"desired-state": "running"```